sympy/sets/fancysets.py

```diff
@@ -659,8 +659,11 @@ def _contains(self, other):
                 _ = self.size  # validate
             except ValueError:
                 return
-        ref = self.start if self.start.is_finite else self.stop
-        if (ref - other) % self.step:  # off sequence
+        if self.start.is_finite:
+            ref = self.start
+        else:
+            ref = self.stop
+        if (ref - other) % self.step is S.true:  # off sequence
```

Comment:
A mod with be a number or nan (or Mod if symbolic values are involved), not S.true.